ui/src/router/components/form/components/router_config/RulesPanel.js

```diff
@@ -13,17 +18,28 @@ import { get } from "../../../../../components/form/utils";
 import { RulesPanelFlyout } from "./RulesPanelFlyout";
 import { RuleCard } from "./rule_card/RuleCard";
 import { useOnChangeHandler } from "../../../../../components/form/hooks/useOnChangeHandler";
-import { newRule } from "../../../../../services/router/TuringRouter";
+import { newDefaultRule, newRule } from "../../../../../services/router/TuringRouter";
 
-export const RulesPanel = ({ rules, routes, onChangeHandler, errors = {} }) => {
+export const RulesPanel = ({ default_traffic_rule, rules, routes, onChangeHandler, rules_errors = {}, default_traffic_rule_errors = {} }) => {
   const { onChange } = useOnChangeHandler(onChangeHandler);
 
   const onAddRule = () => {
+    // Default rule should only be added when there are custom rules
     onChange("rules")([...rules, newRule()]);
+    // rule length check captures the case when all Traffic rules are first loaded
+    // null check captures the case when all Traffic rules are removed and added again
+    if ((rules.length > 0 && !default_traffic_rule) || default_traffic_rule === null) {
```

Comment:
> null check captures the case when all Traffic rules are removed and added again

If they are added again, then we still only need to worry about `rules.length > 0 && !default_traffic_rule` right? Would we ever have a scenario where `rules.length === 0 && default_traffic_rule === null` that we want to handle? And if so, why?

Reply:
`rules.length === 0 && default_traffic_rule === null` - that happens on the first time a Traffic rule is being added. When we first run `onChange("rules")([...rules, newRule()]);` within `onAddRule`, rules that have been passed to the RulesPanel component stays as an empty array, hence `rules.length > 0` will never be true even after clicking the "Add Rule" button.

Reply:
Haha ok.. I have another thought then - if this default rule is created from `onAddRule`, why do we even compare rules length? It's then enough to look at `!default_traffic_rule` only to add the default rule, right?

Reply:
That's a good point, you're right `!default_traffic_rule` is all that's needed, updated to reflect this.